starter-code/src/clue.js

```diff
@@ -1,94 +1,186 @@
+}
 
 // Weapons
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var rope = {
+    name: "rope",
+    weight: 10
+}
+
+var knife = {
+    name: "knife",
+    weight: 8
+}
+
+var candlestick = {
+    name: "candlestick",
+    weight: 2
+}
+
+var dumbbell = {
+    name: "dumbbell",
+    weight: 30
+}
+
+var poison = {
+    name: "poison",
+    weight: 2
+}
+
+var axe = {
+    name: "axe",
+    weight: 15
+}
+
+var bat = {
+    name: "bat",
+    weight: 13
+}
+
+var trophy = {
+    name: "trophy",
+    weight: 25
+}
+
+var pistol = {
+    name: "pistol",
+    weight: 20
+}
+
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
-
-// Characters Collection
-var charactersArray = [];
 
-// Rooms' Collection
-var roomsArray = [];
+var dinningRoom = {
+    name:  "Dinning Room",
+}
+var conservatory = {
+    name: "Conservatory"
+}
+var kitchen = {
+    name: "Kitchen"
+}
+var study = {
+    name: "Study"
+}
+var library = {
+    name: "Library"
+}
+var billiardRoom = {
+    name: "Billiard Room"
+}
+var lounge = {
+    name: "Lounge"
+}
+var ballroom = {
+    name: "Ballroom"
+}
+var hall = {
+    name: "Hall"
+}
+var spa = {
+    name: "Spa"
+}
+var livingRoom = {
+    name: "Living Room"
+}
+var observatory = {
+    name: "Observatory"
+}
+var theater = {
+    name: "Theater"
+}
+var guestHouse = {
+    name: "Guest House"
+}
+var patio ={
+    name: "Patio"
+}
 
-// Weapons Collection
+// Weapons collection:
 var weaponsArray = [];
+
+// Rooms collection:
+var roomsArray = [];
+
+// Characters collection:
+var charactersArray = [];
+
+weaponsArray.push(rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol);
+roomsArray.push(dinningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory,theater,
+guestHouse, patio);
+charactersArray.push(mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard);
+
+
+function randomSelector(category){
+    var card = Math.floor(Math.random()*(category.length));
+    return category[card];
+}
+
+function pickMistery() {
+    var misteryEnvelope =[];
+    misteryEnvelope.push(randomSelector(charactersArray), randomSelector(weaponsArray), randomSelector(roomsArray));
+    return misteryEnvelope;
+}
+
+var mistery= pickMistery();
+console.log(mistery);
+
+function revealMistery (mistery){
+    return (mistery[0].first_name + " " + mistery[0].last_name + " killed Mr. Boddy using the " + mistery[1].name + " in the " + mistery[2].name);
```

Comment:
The only error left on jasmine is here, but it's just a small type error, all your code is good !